lib/puppet/defaults.rb

```diff
@@ -929,17 +929,18 @@ def self.default_vendormoduledir
       :desc  => "Certificate authorities who issue server certificates.  SSL servers will not be
         considered authentic unless they possess a certificate issued by an authority
         listed in this file.  If this setting has no value then the Puppet master's CA
-        certificate (localcacert) will be used."
+        certificate (localcacert) will be used.",
+      :hook => proc do |val|
+        Puppet.deprecation_warning(_("Setting 'ssl_client_ca_auth' is deprecated."))
+      end
     },
     :ssl_server_ca_auth => {
       :type  => :file,
       :mode  => "0644",
       :owner => "service",
       :group => "service",
-      :desc  => "Certificate authorities who issue client certificates.  SSL clients will not be
-        considered authentic unless they possess a certificate issued by an authority
-        listed in this file.  If this setting has no value then the Puppet master's CA
-        certificate (localcacert) will be used."
+      :deprecated  => :completely,
+      :desc => "This setting is deprecated."
```

Comment:
I know it's been dead a while, but do we want to list what the replacement is/was here instead of a plain sentence? Looks like in other places in defaults.rb we've listed what replaced a deprecated setting.

Reply:
Yeah, the motivation for this setting was to support use cases where the client's cert was issued by a different CA or intermediate CA than is contained in 'Puppet[:cacert]'. How about "The setting is deprecated and has no effect. Ensure all certificate authorities used to issue client certificates are contained in the server's CA bundle, contained in the Puppet[:cacert] file on the server."?

Reply:
Updated, I reworded it slightly as "contained in ... contained in" was a bit awkward.